client/lib/browser-storage/index.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * Internal dependencies
+ */
+import {
+	getStoredItem as bypassGet,
+	setStoredItem as bypassSet,
+	clearStorage as bypassClear,
+} from './bypass';
+
+let shouldBypass = false;
+
+const DB_NAME = 'calypso';
+const DB_VERSION = 2; // Match versioning of the previous localforage-based implementation.
+const STORE_NAME = 'calypso_store';
```

Comment:
Can we somehow make this configurable and turn the `lib/browser-storage` into a kind of KV storage polyfill? The `StorageArea` [constructor](https://wicg.github.io/kv-storage/#storagearea-constructor) and the [`get backingStore`](https://wicg.github.io/kv-storage/#storagearea-backingstore) method can serve as inspiration.

Reply:
We could, but I don't think we have a use case for it at the moment. We store everything in the same store, and thus there's no need to configure this. It would also make it harder to have a `localStorage` fallback, as we'd need to emulate multiple stores over `localStorage`, and it would complicate the bypass functionality too.